app/hotels/src/AllHotelsMap/index.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+
+import * as React from 'react';
+import { StyleSheet, View } from 'react-native';
+import { MapView } from 'expo';
+import { PriceMaker } from '@kiwicom/native-common';
+
+import type { HotelMarker, CurrencyInfo } from './../types/datastructures';
+
+const styles = StyleSheet.create({
+  container: {
+    ...StyleSheet.absoluteFillObject,
+  },
+  map: {
+    ...StyleSheet.absoluteFillObject,
+  },
+});
+
+type Props = {
+  markers: HotelMarker[],
+  currencyInfo: CurrencyInfo,
+};
+
+class AllHotelsMap extends React.Component<Props> {
+  map: typeof MapView | null;
+
+  componentDidMount() {
+    // for componentDidMount it's recommended to animate map in next loop for performance
+    setTimeout(() => this.fitToSuppliedMarkers());
+  }
+
+  storeMapReference = (map: React.Ref<typeof MapView> | null) => {
+    this.map = map;
+  };
+
+  fitToSuppliedMarkers(animate = true) {
+    const markerIds = this.props.markers.map(marker => marker.id);
+
+    if (!this.map) {
+      return;
+    }
+
+    this.map.fitToSuppliedMarkers(markerIds, animate);
```

Comment:
Ok, this was really easy... 😄 But it may need additional fiddling:

![simulator screen shot - iphone 8 - 2017-12-07 at 10 07 02](https://user-images.githubusercontent.com/978368/33696827-9e4e2656-db36-11e7-8f03-de19dbc90e09.png)